pubtools/_quay/clear_repo.py

```diff
@@ -47,14 +47,14 @@
         "required": True,
         "type": str,
     },
-    ("--pyxis-krb-principal",): {
-        "help": "Pyxis kerberos principal in form: name@REALM",
+    ("--pyxis-ssl-crtfile",): {
+        "help": "Path to .crt file for the SSL authentication",
         "required": True,
         "type": str,
     },
-    ("--pyxis-krb-ktfile",): {
-        "help": "Pyxis Kerberos client keytab. Optional. Used for login if TGT is not available.",
-        "required": False,
+    ("--pyxis-ssl-keyfile",): {
```

Comment:
Realized while reading the corresponding Pub change:

Why does this command take the cert/key as arguments rather than using the hook?  Is it intentional that some of the commands in pubtools-quay will use the hook to get the cert/key paths, and other commands will instead use command-line arguments?  Why is that?

Reply:
Well, I recall that at one point you mentioned you're disappointed that we use target settings when invoking pubtools quay. So, for the remaining entrypoints I tried to remove dependency on pub (it was also easier, I didn't required instance of hub XMLRPC). Thus, clear and remove repo are implemented as CLI applications, and pub constructs arguments when invoking them. This is unlike other entrypoints invoked by pub, which require some not so CLI friendly parameters (target settings, and instance of hub). When using arguments, pub could resolve the paths and specify them when invoking the new entrypoints. One day we could perhaps unify these approaches.

Reply:
> When using arguments, pub could resolve the paths and specify them when invoking the new entrypoints.

Yeah, it could, but the question is whether it *should* now that the hook exists and is used elsewhere.

Ultimately I think this is a matter for the pubtools-quay maintainers and I don't have a strong opinion either way, but I do think this is being set up less maintainable than it could be, unnecessarily.

You're setting things up so there are two places where Pub is being coupled with this project - it has to *both* implement the hook *and* pass certain arguments into some commands in order for cert/key handling to work properly.  That's two places which need to be maintained or extended/changed if requirements change.  You could easily make it just one place (Pub only has to implement the hook).

Reply:
I thought you wanted to go the other direction. The hook doesn't exactly solve anything since it requires a fallback mechanism (target settings at the moment). Plus, it implies strong coupling with pub. I thought that once we have establish a pub-independent way of passing many parameters we will use that instead of hooks. Since you don't have a strong opinion either way, I won't consider this a blocker.